exec/src/main/scala/cilib/Runner.scala

```diff
@@ -92,33 +111,40 @@ object Runner {
 
               val (r2, next) =
                 e match {
-                  case Unchanged => algorithm.value.run(current).run(newConfig).run(r)
+                  case Unchanged => algorithm.value.run(current).run(state).run(newConfig).run(r)
                   case Change =>
                     val (r3, updated) = onChange(current).run(r)
-                    algorithm.value.run(updated).run(newConfig).run(r3)
+                    algorithm.value.run(updated).run(state).run(newConfig).run(r3)
                 }
 
               next match {
                 case -\/(error) => Process.fail(error)
-                case \/-(value) =>
+                case \/-((newState, value)) =>
                   val progress =
-                    Progress(algorithm.name, problem, r2.seed, iteration, e, value)
-                  Process.emit(progress) ++ go(iteration + 1, r2, value, newConfig)
+                    Progress(algorithm.name, problem, r2.seed, iteration, e, (newState, value))
+                  Process.emit(progress) ++ go(iteration + 1, r2, value, newConfig, newState)
               }
           }
       }
 
     val (rng2, current) = collection.run(rng) // the collection of entities
 
-    env.tee(Process.constant(alg))(go(1, rng2, current, initalConfig))
+    env.tee(alg)(go(1, rng2, current, initialConfig, initialState))
```

Comment:
Say I want to run the same moving peaks but with different initialisation strategies. In the code line 132, rng2 needs to be the same to get the same for all runs to get the same peaks right? Or is the problem's randomness controlled by a separate RNG?

Reply:
Yeah, the problem and the algorithm streams are based off separate RNGs. Otherwise, in the dynamic environment space, the way the algorithm operates (the manner in which it samples RNG values) would impact the environment generation.

Reply:
Ok cool